consensus/p2p/adapters/vote.go

```diff
@@ -0,0 +1,66 @@
+package adapters
+
+import (
+	"errors"
+
+	"github.com/NethermindEth/juno/consensus/starknet"
+	"github.com/NethermindEth/juno/consensus/types"
+	"github.com/NethermindEth/juno/core/address"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/core/hash"
+	"github.com/NethermindEth/juno/utils"
+	"github.com/starknet-io/starknet-p2pspecs/p2p/proto/common"
+	"github.com/starknet-io/starknet-p2pspecs/p2p/proto/consensus/consensus"
+)
+
+type VoteAdapter[H types.Hash, A types.Addr] interface {
+	ToVote(*consensus.Vote) (*types.Vote[H, A], error)
+	FromVote(*types.Vote[H, A], consensus.Vote_VoteType) (*consensus.Vote, error)
+	FromPrevote(*types.Prevote[H, A]) (*consensus.Vote, error)
+	FromPrecommit(*types.Precommit[H, A]) (*consensus.Vote, error)
+}
```

Comment:
All the types being returned here are created inside the function, what do you think of returning them as value types instead?